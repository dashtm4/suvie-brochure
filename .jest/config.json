{
  "rootDir": "../",
  "setupFiles": [
    "<rootDir>/.jest/register-context.js"
  ],
  "moduleNameMapper": {
    "@components/([^\\.]*)$": ["<rootDir>/components/$1"],
    "@components/atoms": ["<rootDir>/components/atoms/index.ts"],
    "@components/molecules": ["<rootDir>/components/molecules/index.ts"],
    "@components/organisms": ["<rootDir>/components/organisms/index.ts"],
    "@components/templates": ["<rootDir>/components/templates/index.ts"],
    "@components/containers": ["<rootDir>/components/containers/index.ts"],
    "@components/sections": ["<rootDir>/components/sections/index.ts"],
    "@hooks": ["<rootDir>/hooks/index.ts"],
    "@queries": ["<rootDir>/queries/index.ts"],
    "@types": ["<rootDir>/types/index.ts"],
    "@utils": ["<rootDir>/utils/index.ts"],
    "@temp/(.*)": [
      "<rootDir>/static/$1",
      "<rootDir>/$1"
    ],
    "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js"
  },
  "setupFilesAfterEnv": [
    "<rootDir>/.jest/setupTests.js"
  ],
  "transform": {
    "^.+\\.(j|t)sx?$": "babel-jest",
    "^.+\\.svg$": "jest-svg-transformer",
    ".+\\.(css|styl|less|sass|scss)$": "jest-css-modules-transform"
  },
  "transformIgnorePatterns": [
    "/node_modules/(?!register-service-worker).+\\.js$"
  ]
}
