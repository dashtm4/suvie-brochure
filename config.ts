export const META_DEFAULTS = {
  title: 'Suvie',
  description: 'Your Countertop Kitchen Robot',
  image: '/images/logo.svg',
  favicon: '/images/favicon.png'
}

export const HELP_EMAIL = 'help@suvie.com'
export const LOGIN_URL = 'https://app.suvie.com/'
export const ACCESSORIES_URL = 'https://store.suvie.com/'
export const HELP_DESC_URL = 'https://support.suvie.com/hc/en-us/'
export const BLOG_URL = 'https://blog.suvie.com'

export const PATHS = {
  BASE: '/',
  KITCHEN_ROBOT: '/kitchen-robot/',
  SMART_MEALS: '/smart-meals/',
  RECIPES: '/recipes/',
  REVIEWS: '/reviews/',
  HEALTH: '/health/',
  ABOUT: '/about-us/',
  PRESS: '/press/',
  JOBS: '/jobs/',
  PRIVACY: '/privacy-policy/',
  TERMS: '/terms-and-conditions/',
  FAQ: '/faq/',
  SHOP_PAGE: '/shop-page/',
  HOW_IT_WORKS: '/how-it-works/',
  FREE_TRIAL: '/free-trial/'
}

export const MAIN_NAV_LINKS: Record<string, any> = [
  {
    title: 'How it Works',
    url: PATHS.KITCHEN_ROBOT
  },
  {
    title: 'Recipes',
    url: PATHS.RECIPES
  },
  {
    title: 'Meal Plans',
    url: PATHS.SMART_MEALS
  },
  {
    title: 'Reviews',
    url: PATHS.REVIEWS
  },
  {
    title: 'Health',
    url: PATHS.HEALTH
  }
]

export const FOOTER_MENU_1 = [
  {
    title: 'Accessories',
    url: ACCESSORIES_URL,
    target: '_self',
    rel: '',
    external: true
  },
  {
    title: 'Log In',
    url: LOGIN_URL,
    external: true
  },
  {
    title: 'How it Works',
    url: PATHS.KITCHEN_ROBOT
  },
  {
    title: 'Recipes',
    url: PATHS.RECIPES
  },
  {
    title: 'Meal Plans',
    url: PATHS.SMART_MEALS
  },
  {
    title: 'Reviews',
    url: PATHS.REVIEWS
  },
  {
    title: 'Health',
    url: PATHS.HEALTH
  }
]

export const FOOTER_MENU_2 = [
  {
    title: 'About',
    url: PATHS.ABOUT
  },
  /* {
    title: 'Press',
    url: PATHS.PRESS
  }, */
  /* {
    title: 'Jobs',
    url: PATHS.JOBS
  }, */
  {
    title: 'Free Trial',
    url: PATHS.FREE_TRIAL
  },
  {
    title: 'Privacy',
    url: PATHS.PRIVACY
  },
  {
    title: 'Terms',
    url: PATHS.TERMS
  }
]

export const FOOTER_MENU_3 = [
  {
    title: 'Help Desk',
    url: HELP_DESC_URL,
    external: true
  },
  {
    title: 'FAQ',
    url: PATHS.FAQ
  }
]

export const FOOTER_COPY = `Meals for purchase may vary based on availability. Time savings are estimates and not a
guarantee. Some images simulated or sped up and some servings suggested with finishing steps and consumer supplied additional
ingredients. Meal coupon offers do not apply to the Suvie appliance.  For all coupons unless otherwise specified
only one coupon is valid per customer for one-time use only. Coupons are valid only during their coupon period.
Sales tax, shipping & handling do not qualify for any coupons. Coupons are not redeemable for cash or gift cards.
Coupons are not valid for prior purchases. Coupons may not be combined with any other coupons, discounts, offers, or
promotions. Void if altered, copied, transferred, auctioned or sold. Offer valid on in-stock items only. No rain checks
issued. Other restrictions may apply. *Additional shipping and handling charges may apply see terms and conditions
for details. Features subject to change on final product. Patents Pending. Copyright 2017 -
  ${new Date().getFullYear()}, Suvie.`
