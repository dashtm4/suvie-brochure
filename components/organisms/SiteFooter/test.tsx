import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { FOOTER_COPY } from '@temp/config'

import { SiteFooter } from '.'

describe('<SiteFooter />', () => {
  it('should renders correct <SiteFooter />', () => {
    const wrapper = shallow(<SiteFooter />)

    expect(wrapper.contains(FOOTER_COPY)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
