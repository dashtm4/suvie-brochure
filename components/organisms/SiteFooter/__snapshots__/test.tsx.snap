// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SiteFooter /> should renders correct <SiteFooter /> 1`] = `
<footer
  className="relative bg-secondary mt-auto mb-0 pt-6 pb-5 md:pt-9 md:pb-12"
>
  <Wave />
  <Container>
    <Row
      className="mb-8"
    >
      <Col
        className="w-full md:w-4/12 mb-10 md:mb-0 md:order-1"
      >
        <KlaviyoSubscriptionForm />
        <SocialMediaList
          list={
            Array [
              Object {
                "icon": "/images/facebook-icon.svg",
                "title": "Facebook",
                "url": "https://www.facebook.com/SuvieFood/",
              },
              Object {
                "icon": "/images/twitter-icon.svg",
                "title": "Twitter",
                "url": "https://twitter.com/suviefood",
              },
              Object {
                "icon": "/images/instagram-icon.svg",
                "title": "Instagram",
                "url": "https://www.instagram.com/suviefood/",
              },
              Object {
                "icon": "/images/pinterest-icon.svg",
                "title": "Pinterest",
                "url": "https://www.pinterest.com/suviefood/",
              },
            ]
          }
        />
      </Col>
      <Col
        className="md:w-8/12 md:order-0"
      >
        <Row>
          <Col
            className="w-full md:w-1/5 mb-10 md:mb-0"
          >
            <SealLogo />
          </Col>
          <Col
            className="w-1/2 md:w-1/5 mb-4"
          >
            <FooterMenuTitle>
              Suvie
            </FooterMenuTitle>
            <FooterMenu
              menu={
                Array [
                  Object {
                    "external": true,
                    "rel": "",
                    "target": "_self",
                    "title": "Accessories",
                    "url": "https://store.suvie.com/",
                  },
                  Object {
                    "external": true,
                    "title": "Log In",
                    "url": "https://app.suvie.com/",
                  },
                  Object {
                    "title": "How it Works",
                    "url": "/kitchen-robot/",
                  },
                  Object {
                    "title": "Recipes",
                    "url": "/recipes/",
                  },
                  Object {
                    "title": "Meal Plans",
                    "url": "/smart-meals/",
                  },
                  Object {
                    "title": "Reviews",
                    "url": "/reviews/",
                  },
                  Object {
                    "title": "Health",
                    "url": "/health/",
                  },
                ]
              }
            />
          </Col>
          <Col
            className="w-1/2 md:w-1/5 mb-4"
          >
            <FooterMenuTitle>
              Company
            </FooterMenuTitle>
            <FooterMenu
              menu={
                Array [
                  Object {
                    "title": "About",
                    "url": "/about-us/",
                  },
                  Object {
                    "title": "Free Trial",
                    "url": "/free-trial/",
                  },
                  Object {
                    "title": "Privacy",
                    "url": "/privacy-policy/",
                  },
                  Object {
                    "title": "Terms",
                    "url": "/terms-and-conditions/",
                  },
                ]
              }
            />
          </Col>
          <Col
            className="w-1/2 md:w-1/5 mb-4"
          >
            <FooterMenuTitle>
              Support
            </FooterMenuTitle>
            <FooterMenu
              menu={
                Array [
                  Object {
                    "external": true,
                    "title": "Help Desk",
                    "url": "https://support.suvie.com/hc/en-us/",
                  },
                  Object {
                    "title": "FAQ",
                    "url": "/faq/",
                  },
                ]
              }
            />
          </Col>
        </Row>
      </Col>
    </Row>
    <p
      className="text-xs text-gray-600 "
    >
      Meals for purchase may vary based on availability. Time savings are estimates and not a
guarantee. Some images simulated or sped up and some servings suggested with finishing steps and consumer supplied additional
ingredients. Meal coupon offers do not apply to the Suvie appliance.  For all coupons unless otherwise specified
only one coupon is valid per customer for one-time use only. Coupons are valid only during their coupon period.
Sales tax, shipping & handling do not qualify for any coupons. Coupons are not redeemable for cash or gift cards.
Coupons are not valid for prior purchases. Coupons may not be combined with any other coupons, discounts, offers, or
promotions. Void if altered, copied, transferred, auctioned or sold. Offer valid on in-stock items only. No rain checks
issued. Other restrictions may apply. *Additional shipping and handling charges may apply see terms and conditions
for details. Features subject to change on final product. Patents Pending. Copyright 2017 -
  2021, Suvie.
    </p>
  </Container>
</footer>
`;
